python/tests/operations/join.py

```diff
@@ -275,6 +275,55 @@ def test_different_names(self):
 
         self.assertTrue(np.equal(joined_labels.tolist(), ref).all())
 
+    def test_different_names_different_length(self):
+        """Test Label joining using labels with different names and different length."""
+        property_labels_1 = Labels(
+            ("structure", "prop_1"), np.vstack(2 * [np.arange(5)]).T
+        )
+        property_labels_2 = Labels(
+            ("structure", "prop_2", "prop_3"), np.vstack(3 * [np.arange(5)]).T
+        )
+
+        block_1 = TensorBlock(
+            values=np.zeros([2, 5]),
+            samples=self.sample_labels,
+            components=[],
+            properties=property_labels_1,
+        )
+
+        block_2 = TensorBlock(
+            values=np.zeros([2, 5]),
+            samples=self.sample_labels,
+            components=[],
+            properties=property_labels_2,
+        )
+
+        joined_ts = eop.join(
```

Comment:
what does `joined_ts` mean? shouldn't be `joined_tm`

Reply:
would prefer for readability `equi_ops`, but not important for this PR

Reply:
Both fine for me. I change this in a next commit.

Reply:
> would prefer for readability equi_ops, but not important for this PR

at this point, why not optimize for reading instead of writing and go for `equistore.operations.join`? Any of you feeling strongly about this? I don't mind much

Reply:
No strong feelings, only eop is hard to interpret for me without reading the import

Reply:
I am fine with using the extended version.